src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/Lobby.java

```diff
@@ -113,6 +116,32 @@ public String getInvitationCode() {
         return this.invitationCode;
     }
 
+    @Override
+    public void setUserName(String token, String newName) {
+        for (Player player: playerMap.values())
+            if (player.getName().equals(newName))
+                throw new ResponseStatusException(HttpStatus.CONFLICT, "Username " + newName + " is already taken.");
```

Comment:
ResponseStatusException should be thrown in the LobbyService not in the Lobby.

Reply:
Done. May I ask why the Service and not the Controller?

Reply:
Separation of Concerns. The controller is the (more or less) interface to the outside world and nothing more. (And that's how they provided the template).
It makes the life also easier for us. Whenever we find some ResponseStatusException, we know, we have to go to the Service and look what's happening there...

Reply:
thanks